src/workflow/Util.hpp

```diff
@@ -35,6 +38,16 @@ namespace workflow {
     bool mergeOutputTableSummaryReports(IdfObject& existingObject, const IdfObject& newObject);
     bool addEnergyPlusOutputRequest(Workspace& workspace, IdfObject& idfObject);
 
+    // Schedule can be used onto a meter/variable, the EMS/PythonPlugin **could** be used for custom reporting
+    // Also adding some cost/utility related objects, the goal is to be not too restrictive here
+    static constexpr std::array<std::string_view, 11> safe_idd_prefixes{
+      "Output",     "Meter",    "PythonPlugin", "EnergyManagementSystem", "Schedule", "LifeCycleCost", "UtilityCost", "ComponentCost",
+      "Compliance", "Currency", "FuelFactors",
+    };
+
+    // If the IddObjectType.valueName doesn't start with one of the allowed prefixes (see `safe_idd_prefixes`), it returns true.
+    bool isEnergyPlusOutputRequestPotentiallyUnsafe(const IddObjectType& iddObjectType);
```

Comment:
* I aimed to strike the right balance here: the list is not restrictive on purpose
* Remember that this is just resulting in a Warn message, it'll still process your request.
* Also, energyplusOutputRequests is somewhat moot now that we have a modelOutputRequests method
* Finally, I made the list of prefix really easy to change: just add to that std::array and you're good to go, which would allow anyone to propose a change even without knowing much C++